javagen/src/main/java/com/azure/autorest/mapper/ProxyMethodMapper.java

```diff
@@ -78,6 +81,15 @@ public ProxyMethod map(Operation operation) {
         }
         IType returnType = GenericType.Mono(singleValueType);
 
+        // method with schema in headers would require a ClientResponse
+        if (operation.getResponses().stream()
+                .filter(r -> r.getProtocol() != null && r.getProtocol().getHttp() != null && r.getProtocol().getHttp().getHeaders() != null)
+                .flatMap(r -> r.getProtocol().getHttp().getHeaders().stream().map(Header::getSchema))
+                .anyMatch(Objects::nonNull)) {
+            final String clientResponseName = operation.getOperationGroup().getLanguage().getJava().getName() + CodeNamer.toPascalCase(operation.getLanguage().getJava().getName()) + "Response";
+            returnType = GenericType.Mono(new ClassType(settings.getPackage(settings.getModelsSubpackage()), clientResponseName));
+        }
+
```

Comment:
This part is of same logic as ClientMapper.ParseResponse. If OK I would refactor to avoid duplicate.

Reply:
Feel free to make local refactorings.

Reply:
Refactor out class name and package name.